File: Basics/Patterns.py
def pattern1(n):
    print('Pattern 1')
    print('--------------------')
    for i in range(1,n+1):
        for j in range(1,n+1):
            print('*',end='')
        print()
        
def pattern2(n):
    print('Pattern 2')
    print('--------------------')
    for i in range(1,n+1):
        print('*'*i)
        
def pattern3(n):
    print('Pattern 3')
    print('--------------------')
    for i in range(1,n+1):
        for j in range(i):
            print(j+1,end='')
        print()
        
def pattern4(n):
    print('Pattern 4')
    print('--------------------')
    for i in range(1,n+1):
        for j in range(i):
            print(i,end='')
        print()
        
def pattern5(n):
    print('Pattern 5')
    print('--------------------')
    for i in range(1,n+1):
        for j in range(n-i+1):
            print('*',end='')
        print()
        
def pattern6(n):
    print('Pattern 6')
    print('--------------------')
    for i in range(1,n+1):
        for j in range(1,n-i+2):
            print(j,end='')
        print()
        
def pattern7(n):
    print('Pattern 7')
    print('--------------------')
    for i in range(1,n+1):
        for j in range(n-i):
            print(' ',end='')
        for j in range(2*i-1):
            print('*',end='')
        for j in range(n-i):
            print(' ',end='')
        print()
        
def pattern8(n):
    print('Pattern 8')
    print('--------------------')
    for i in range(1,n+1):
        for j in range(i-1):
            print(' ',end='')
        for j in range(2*(n-i)+1):
            print('*',end='')
        for j in range(i-1):
            print(' ',end='')
        print()
        
def pattern9(n):
    print('Pattern 9')
    print('--------------------')
    for i in range(1,n+1):
        for j in range(n-i):
            print(' ',end='')
        for j in range(2*i-1):
            print('*',end='')
        for j in range(n-i):
            print(' ',end='')
        print()
        
    for i in range(1,n+1):
        for j in range(i-1):
            print(' ',end='')
        for j in range(2*(n-i)+1):
            print('*',end='')
        for j in range(i-1):
            print(' ',end='')
        print()
        
def pattern10(n):
    print('Pattern 10')
    print('--------------------')
    for i in range(1,n+1):
        print('*'*i)
    for i in range(1,n):
        for j in range(n-i):
            print('*',end='')
        print()
        
def pattern11(n):
    print('Pattern 11')
    print('--------------------')
    for i in range(1,n+1):
        start = 1
        if i%2==0:
            start = 0
        for j in range(i):
            print(start,end=' ')
            start = 1 - start
        print()
        
def pattern12(n):
    print('Pattern 12')
    print('--------------------')
    for i in range(1,n+1):
        for j in range(i):
            print(j+1,end='')
        for j in range(2*(n-i)):
            print(' ',end='')
        for j in range(i,0,-1):
            print(j,end='')
        print()
        
def pattern13(n):
    print('Pattern 13')
    print('--------------------')
    num = 1
    for i in range(1,n+1):
        for j in range(i):
            print(num,end=' ')
            num = num + 1
        print()
        
def pattern14(n):
    print('Pattern 14')
    print('--------------------')
    letter = 'A'
    for i in range(1,n+1):
        for j in range(i):
            print(chr(ord(letter)+j),end='')
        print()
    
def pattern15(n):
    print('Pattern 15')
    print('--------------------')
    letter = 'A'
    for i in range(1,n+1):
        for j in range(n-i+1):
            print(chr(ord(letter)+j),end='')
        print()

def pattern16(n):
    print('Pattern 16')
    print('--------------------')
    letter = 'A'
    for i in range(1,n+1):
        for j in range(i):
            print(chr(ord(letter)+i-1),end='')
        print()
        
def pattern17(n):
    print('Pattern 17')
    print('--------------------')
    letter = 'A'
    for i in range(1,n+1):
        for j in range(n-i):
            print(' ',end='')
        for j in range(i):
            print(chr(ord(letter)+j),end='')
        for j in range(i-1):
            print(chr(ord(letter)+i-j-2),end='')
        print()
        
def pattern18(n):
    print('Pattern 18')
    print('--------------------')
    letter = 'E'
    for i in range(1,n+1):
        for j in range(i):
            print(chr(ord(letter)-i+j+1),end=' ')
        print()
        
def pattern19(n):
    print('Pattern 19')
    print('--------------------')
    for i in range(1,n+1):
        for j in range(n-i+1):
            print('*',end='')
        for j in range(2*(i-1)):
            print(' ',end='')
        for j in range(n-i+1):
            print('*',end='')
        print()
        
    for i in range(1,n+1):
        for j in range(i):
            print('*',end='')
        for j in range(2*(n-i)):
            print(' ',end='')
        for j in range(i):
            print('*',end='')
        print()
        
def pattern20(n):
    print('Pattern 20')
    print('--------------------')
    for i in range(1,n+1):
        for j in range(i):
            print('*',end='')
        for j in range(2*(n-i)):
            print(' ',end='')
        for j in range(i):
            print('*',end='')
        print()
        
    for i in range(1,n):
        for j in range(n-i):
            print('*',end='')
        for j in range(2*i):
            print(' ',end='')
        for j in range(n-i):
            print('*',end='')
        print()
        
def pattern21(n):
    print('Pattern 21')
    print('--------------------')
    for i in range(1,n+1):
        for j in range(1,n+1):
            if i==n or j==n or i==1 or j==1:
                print('*',end='')
            else:
                print(' ',end='')
        print()
        
def pattern22(n):
    print('Patter 22')
    print('--------------------')
    p = 2*n - 1
    for i in range(p):
        for j in range(p):
            top = i
            left = j
            right = 2*n - 2 - j
            bottom = 2*n - 2 - i 
            res = n - min(min(top,bottom),min(right,left))
            print(res,end='')
        print()

patterns = [pattern1, pattern2, pattern3, pattern4, pattern5, pattern6, pattern7, pattern8, pattern9, pattern10, pattern11, pattern12, pattern13, pattern14, pattern15, pattern16, pattern17, pattern18, pattern19, pattern20, pattern21,pattern22]

for pattern in patterns:
    print('--------------------')
    pattern(5)
print()

File: Basics/Patterns.java
package Basics;

public class Patterns {
    public static void pattern1(int n) {
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                System.out.print('*');
            }
            System.out.println();
        }
    }
    public static void pattern2(int n) {
        for(int i=0;i<n;i++){
            for(int j=0;j<=i;j++){
                System.out.print('*');
            }
            System.out.println();
        }
    }
    public static void pattern3(int n) {
        for(int i=0;i<n;i++){
            for(int j=0;j<=i;j++){
                System.out.print(j+1);
            }
            System.out.println();
        }
    }
    public static void pattern4(int n) {
        for(int i=0;i<n;i++){
            for(int j=0;j<=i;j++){
                System.out.print(i+1);
            }
            System.out.println();
        }
    }
    public static void pattern5(int n) {
        for(int i=0;i<n;i++){
            for(int j=i;j<n;j++){
                System.out.print('*');
            }
            System.out.println();
        }
    }
    public static void pattern6(int n) {
        for(int i=0;i<n;i++){
            for(int j=0;j<n-i;j++){
                System.out.print(j+1);
            }
            System.out.println();
        }
    }
    public static void pattern7(int n) {
        for(int i=0;i<n;i++){
            for(int j=0;j<n-i-1;j++){
                System.out.print(" ");
            }
            for(int j=0;j<2*i+1;j++){
                System.out.print("*");
            }
            for(int j=0;j<n-i;j++){
                System.out.print(" ");
            }
            System.out.println();
        }
    }
    public static void pattern8(int n) {
        for(int i=0;i<n;i++){
            for(int j=0;j<i;j++){
                System.out.print(" ");
            }
            for(int j=0;j<2*(n-i)-1;j++){
                System.out.print("*");
            }
            for(int j=0;j<i;j++){
                System.out.print(" ");
            }
            System.out.println();
        }
    }
    public static void pattern9(int n) {
        for(int i=0;i<n;i++){
            for(int j=0;j<n-i-1;j++){
                System.out.print(" ");
            }
            for(int j=0;j<2*i+1;j++){
                System.out.print("*");
            }
            for(int j=0;j<n-i;j++){
                System.out.print(" ");
            }
            System.out.println();
        }
        for(int i=0;i<n;i++){
            for(int j=0;j<i;j++){
                System.out.print(" ");
            }
            for(int j=0;j<2*(n-i)-1;j++){
                System.out.print("*");
            }
            for(int j=0;j<i;j++){
                System.out.print(" ");
            }
            System.out.println();
        }
    }
    public static void pattern10(int n) {
        for(int i=0;i<n;i++){
            for(int j=0;j<i+1;j++){
                System.out.print('*');
            }
            System.out.println();
        }
        for(int i=0;i<n-1;i++){
            for(int j=0;j<n-i-1;j++){
                System.out.print('*');
            }
            System.out.println();
        }
    }
    public static void pattern11(int n) {
        for(int i=0;i<n;i++){
            int start = 1;
            if(i%2==1){
                start = 0;
            }
            for(int j=0;j<i+1;j++){
                System.out.print(start+" ");
                start = 1 - start;
            }
            System.out.println();
        }
    }
    public static void pattern12(int n) {
        n = n-1;
        for(int i=0;i<n;i++){
            for(int j=0;j<i+1;j++){
                System.out.print(j+1);
            }
            for(int j=0;j<2*(n-i-1);j++){
                System.out.print(" ");
            }
            for(int j=i+1;j>0;j--){
                System.out.print(j);
            }
            System.out.println();
        }
    }
    public static void pattern13(int n) {
        int num = 1;
        for(int i=0;i<n;i++){
            for(int j=0;j<i+1;j++){
                System.out.print(num+" ");
                num++;
            }
            System.out.println();
        }
    }
    public static void pattern14(int n){
        char ch = 'A';
        for(int i=0;i<n;i++){
            for(int j=0;j<i+1;j++){
                System.out.print((char)(ch+j));
            }
            System.out.println();
        }
    }
    public static void pattern15(int n){
        char ch = 'A';
        for(int i=0;i<n;i++){
            for(int j=0;j<n-i;j++){
                System.out.print((char)(ch+j));
            }
            System.out.println();
        }
    }
    public static void pattern16(int n){
        char ch = 'A';
        for(int i=0;i<n;i++){
            for(int j=0;j<i+1;j++){
                System.out.print((char)(ch+i));
            }
            System.out.println();
        }
    }
    public static void pattern17(int n){
        char ch = 'A';
        n = n-1;
        for(int i=0;i<n;i++){
            for(int j=0;j<n-i-1;j++){
                System.out.print(" ");
            }
            for(int j=0;j<i+1;j++){
                System.out.print((char)(ch+j));
            }
            for(int j=0;j<i;j++){
                System.out.print((char)(ch+i-j-1));
            }
            System.out.println();
        }
    }
    public static void pattern18(int n){
        char ch = 'E';
        for(int i=0;i<n;i++){
            for(int j=0;j<i+1;j++){
                System.out.print((char)(ch-i+j)+" ");
            }
            System.out.println();
        }
    }
    public static void pattern19(int n){
        for(int i=0;i<n;i++){
            for(int j=0;j<n-i;j++){
                System.out.print("*");
            }
            for(int j=0;j<2*i;j++){
                System.out.print(" ");
            }
            for(int j=0;j<n-i;j++){
                System.out.print("*");
            }
            System.out.println();
        }
        for(int i=0;i<n;i++){
            for(int j=0;j<i+1;j++){
                System.out.print("*");
            }
            for(int j=0;j<2*(n-i-1);j++){
                System.out.print(" ");
            }
            for(int j=0;j<i+1;j++){
                System.out.print("*");
            }
            System.out.println();
        }
    }
    public static void pattern20(int n){
        for(int i=0;i<n;i++){
            for(int j=0;j<i+1;j++){
                System.out.print("*");
            }
            for(int j=0;j<2*(n-i-1);j++){
                System.out.print(" ");
            }
            for(int j=0;j<i+1;j++){
                System.out.print("*");
            }
            System.out.println();
        }
        for(int i=0;i<n-1;i++){
            for(int j=0;j<n-i-1;j++){
                System.out.print("*");
            }
            for(int j=0;j<2*(i+1);j++){
                System.out.print(" ");
            }
            for(int j=0;j<n-i-1;j++){
                System.out.print("*");
            }
            System.out.println();
        }
    }
    public static void pattern21(int n){
        n = n-1;
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                if(i==0||j==0||i==n-1||j==n-1){
                    System.out.print('*');
                } else {
                    System.out.print(' ');
                }
            }
            System.out.println();
        }
    }
    public static void pattern22(int n){
        n = n-1;
        int boundary = 2*n-1;
        for(int i=0;i<boundary;i++){
            for(int j=0;j<boundary;j++){
                // Distance from edges
                int top = i, bottom = boundary-1-i;
                int left = j, right = boundary-1-j;
                // Find the minimum distance from edges and value = n - min distance
                int res = n - Math.min(Math.min(top,bottom),Math.min(right,left));
                System.out.print(res+"");
            }
            System.out.println();
        }
    }
    public static void main(String[] args) {
        int n = 5;
        System.out.println("Pattern 1:");
        pattern1(n);
        System.out.println("Pattern 2:");
        pattern2(n);
        System.out.println("Pattern 3:");
        pattern3(n);
        System.out.println("Pattern 4:");
        pattern4(n);
        System.out.println("Pattern 5:");
        pattern5(n);
        System.out.println("Pattern 6:");
        pattern6(n);
        System.out.println("Pattern 7:");
        pattern7(n);
        System.out.println("Pattern 8:");
        pattern8(n);
        System.out.println("Pattern 9:");
        pattern9(n);
        System.out.println("Pattern 10:");
        pattern10(n);
        System.out.println("Pattern 11:");
        pattern11(n);
        System.out.println("Pattern 12:");
        pattern12(n);
        System.out.println("Pattern 13:");
        pattern13(n);
        System.out.println("Pattern 14:");
        pattern14(n);
        System.out.println("Pattern 15:");
        pattern15(n);
        System.out.println("Pattern 16:");
        pattern16(n);
        System.out.println("Pattern 17:");
        pattern17(n);
        System.out.println("Pattern 18:");
        pattern18(n);
        System.out.println("Pattern 19:");
        pattern19(n);
        System.out.println("Pattern 20:");
        pattern20(n);
        System.out.println("Pattern 21:");
        pattern21(n);
        System.out.println("Pattern 22:");
        pattern22(n);
    }
}
